### PR TITLE
Add support replacement with nodes

### PR DESCRIPTION
**Pull request by rtsao**
This is useful for replacing variables with non-boolean and non-string values.

### CODE REVIEW

File: test/test.js [hunk 1]
```diff
@@ -33,6 +34,33 @@ if (true) {
 
 });
 
+describe('node replacement', () => {
+    describe(`transform`, () => {
+        const nodeEnv = t.memberExpression(t.memberExpression(t.identifier('process'), t.identifier('env')), t.identifier('NODE_ENV'));
+        const nodeEnvCheck = t.binaryExpression('!==', nodeEnv, t.stringLiteral('production'));
+
```
Comment (wssgcg1213) [inline test/test.js:41]: I think it's not a friendly api for users to understand babel node or babel-types . maybe we can make it easier to use. like 
```
plugins: [[plugin, {
   __DEV__: {
    type: 'node', // default to string
    replacement: 'process.env.NODE_ENV'
  }
}]]
```

**Comment by wssgcg1213** [comment]
I find a way to do this.

image  

image

it was replaced by `process.env.NODE_ENV` not `'process.env.NODE_ENV'` (String)  

any ideas?

**Comment by rtsao** [comment]
There's actually native support for this in babel plugins, see:
https://github.com/thejameskyle/babel-handbook/blob/master/translations/en/plugin-handbook.md#replacing-a-node-with-a-source-string

I didn't like the idea of re-parsing strings repeatedly so I opted to accept babel nodes directly. Of course the performance overhead could be mitigated by memoization, but I don't think the following is overly burdensome:
```js
plugins: [[plugin, {
   __DEV__: babylon.parse('process.env.NODE_ENV')
}]]
```

**Comment by wssgcg1213** [comment]
Pass a babel node directly is acceptable for this plugin to support.    
  
My point is users may donnot understand Node, AST, babylon or babel-types. We should provide an another easier way for these users to use this plugin.

**Pull request closed without merging**